tests/integration/framework/process/helmtpl/helm.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helmtpl
+
+import (
+	"bytes"
+	"context"
+	"os"
+	oexec "os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework/binary"
+)
+
+// Helm test helm chart template.  It is not really an integration test but this seems the best place to place it
+type Helm struct {
+	defaultOpts options
+}
+
+func New(t *testing.T, fopts ...Option) *Helm {
+	t.Helper()
+
+	var opts options
+
+	for _, fopt := range fopts {
+		fopt(&opts)
+	}
+
+	return &Helm{
+		defaultOpts: opts,
+	}
+}
+
+func (h *Helm) Render(t *testing.T, ctx context.Context, fopts ...Option) []byte {
+	t.Helper()
+
+	var opts options
+
+	for _, fopt := range fopts {
+		fopt(&opts)
+	}
+
+	if runtime.GOOS == "darwin" && os.Getenv("GITHUB_ACTIONS") == "true" {
+		t.Skip("Skipping helm tests on macos in Github Actions. As Helm is not present in the macos runner.")
```

Comment:
The only dependency for integration tests is strictly `go` so we can't call out to a helm binary.

Reply:
ok. there are a couple options here as well. I think using helm directly provides the most accurate representation of what a user does, but, other options are:
- we could also use the helm api that is in Go (I might remove some options related to set values, like json, string, etc). This will inject helm dependencies into go.mod 
- only run if helm binary is present
- move out of integration to a place where this guarantee is not present. we lose some of the nice things added to integration tests

Reply:
Another option is to create a `main.go` with a separate `go.mod` to prevent needing the helm import at the top level. We are doing a similar thing [here](https://github.com/dapr/dapr/pull/7575/files#diff-29f7f056025f5cf907b18b183bc69d804c2c101c65c0cd47072f3a31cf0f19f4).